### DISCUSSION TITLE
Let's fix packages that are broken in the automatic package updater

### DISCUSSION THREAD

**Discussion by finagolfin** [termux/termux-packages#7393] [Announcements]
We currently have [an automatic package updater](https://github.com/termux/termux-packages/blob/master/scripts/updates/check-updates.sh), that [runs four times a day](https://github.com/termux/termux-packages/blob/master/.github/workflows/package_updates.yml) on [a list of github projects](https://github.com/termux/termux-packages/blob/master/scripts/updates/github-projects.txt). The updates it adds can be found in [the recent commits as done by `Termux Github Actions`](https://github.com/termux/termux-packages/commits/master).

However, it currently fails for 15-20 packages because of some issues with the patches getting outdated or the updated build breaking. You can see which packages are not getting updated by going to [`Actions` and clicking on the most recent one saying `Package updates`](https://github.com/termux/termux-packages/actions), then checking its log.

Let's fix these, because they will usually stay automatically updated from then on.

**Comment by thunder-coding** [comment]
Here are the packages I found which are failing to be build:

- android-tools
~~- ccache~~
- enchant
- erlang (I am already working on this in #7073)
- et
- gitea
- i2pd
- iverilog
- keybase
- libprotobuf-c
- librav1e
- loksh
- mu
- nushell
- proxychains-ng
- qpdf
- zoxide

**Reply by ghost** [reply]
Don't update `htop-legacy`. That's why it specifically called `htop` "legacy", i.e. old version branch.

Updated variant is in root-repo.

**Reply by finagolfin** [reply]
Heh, then perhaps it shouldn't be in [the list of packages to be updated](https://github.com/termux/termux-packages/blob/master/scripts/updates/github-projects.txt#L136). 😄

**Reply by finagolfin** [reply]
ccache and mu fixed, pulls open for erlang and et.

**Reply by finagolfin** [reply]
android-tools update depends on header added to NDK 23.

**Comment by finagolfin** [comment]
@xtkoba, nice work adding so many packages lately. 😃 These updates have fallen even more behind, if you have the time to look at this.

**Comment by xtkoba** [comment]
Please file issues, one per package. Then I will look into them.

**Reply by finagolfin** [reply]
It is easier to simply follow the instructions I wrote above, to see what packages are breaking on the auto-updater. If you prefer to add new packages instead, no problem.

**Comment by mcfriend99** [comment]
@xtkoba thanks for adding Blade. Just wanted to let you know that Blade has since moved from version `0.0.5` to version `0.0.7`. Would appreciate having the updated version on my Android phone.
